src/tools/illink/test/Mono.Linker.Tests.Cases/Reflection/ObjectGetType.cs

```diff
@@ -1188,7 +1188,7 @@ class FieldTypeAlsoImplementsInterface : IAnnotatedInterface
 				public NestedFieldType _nestedField;
 
 				[Kept]
-				public class NestedFieldType
+				public struct NestedFieldType
```

Comment:
Native AOT will still try to optimize out the MethodTable of a type even if it's used as the type of a reflection-visible field so this wasn't kept. It is not able to optimize it out if the type of the field is a struct since reflection APIs (FieldInfo.GetValue) will create a default-initialized boxed version when called.